src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,78 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userData, setUserData] = useState<User | null>(null);
+  const [hasErrorMessage, setErrorMessage] = useState('');
```

Comment:
The state variable `hasErrorMessage` is misleadingly named. Since it holds a string message rather than a boolean, consider renaming it to `errorMessage` to better reflect its purpose.